controllers/btpoperator_controller.go

```diff
@@ -285,13 +287,37 @@ func (r *BtpOperatorReconciler) HandleRedundantCR(ctx context.Context, oldestCr
 }
 
 func (r *BtpOperatorReconciler) UpdateBtpOperatorStatus(ctx context.Context, cr *v1alpha1.BtpOperator, newState v1alpha1.State, reason conditions.Reason, message string) error {
-	cr.Status.WithState(newState)
-	newCondition := conditions.ConditionFromExistingReason(reason, message)
-	if newCondition != nil {
-		conditions.SetStatusCondition(&cr.Status.Conditions, *newCondition)
+	logger := log.FromContext(ctx)
+	timeout := time.Now().Add(StatusUpdateTimeout)
+
+	var err error
+	for now := time.Now(); now.Before(timeout); now = time.Now() {
+		if err = r.Get(ctx, client.ObjectKeyFromObject(cr), cr); err != nil {
+			if k8serrors.IsNotFound(err) {
+				return nil
+			}
+			logger.Error(err, "cannot get the BtpOperator to update the status. Retrying in 500 ms...")
```

Comment:
Can the "500" literal be changed to "StatusUpdateCheckInterval" const?

Reply:
suggestion applied.